### DISCUSSION TITLE
[SOLVED] @aws-cdk/aws-lambda-python-alpha: Use custom bundling image from ECR registry

### DISCUSSION THREAD

**Discussion by dobeerman** [aws/aws-cdk#23914] [General]
### Describe the issue

We use a bunch of python lambdas the build process of takes too much time. To speed up the process we want to use custom image that already includes all dependencies for that lambdas. To achieve such a great feature I try to specify the image as follow:
```ts
bundling: {
  image: cdk.DockerImage.fromRegistry(
    `${props.env.account}.dkr.ecr.${props.env.region}.amazonaws.com/com-python-build-image:latest`,
  ),
},
```
Unfortunatelly, I got the error 
```
Bundling asset Root-dev/...[CUT].../Code/Stage...
Unable to find image '[123456789012.dkr.ecr.eu-central-1.amazonaws.com/com-python-build-image:latest](http://123456789012.dkr.ecr.eu-central-1.amazonaws.com/com-python-build-image:latest)' locally
docker: Error response from daemon: Head "https://123456789012.dkr.ecr.eu-central-1.amazonaws.com/v2/com-python-build-image/manifests/latest": no basic auth credentials.
```
Both cdk and ecr repo are in the same aws account and I use OIDC authorization.

What I do wrong? :thinking: 

### Links

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.DockerImage.html

**Comment by dobeerman** [comment]
So, I had to login `aws ecr get-login-password --region ${AWS_REGION} | docker login --username AWS --password-stdin ${ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com` to be able to download the image.

:question: Can anyone give an idea how to put that image we use for bundling inside of another image we use in pipeline?

**Comment by pahud** [comment]
Hi, can you elaborate a little bit more about what you are trying to do?

AFAIK, I believe you can define your custom dockerfile in the path passed to DockerImage.fromBuild() so you have a custom docker image for custom bundling.
https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk/aws-lambda-python#custom-bundling

I am moving this to discussion as this is not relevant to a bug or feature request.

**Reply by dobeerman** [reply]
Hi Pahud,
Thanks for the swift reply.

We use Bitbucket pipeline to deploy the stack. The stack consists of bunch of python lambdas we build with the `@aws-cdk/aws-lambda-python-alpha` construct. Every time the pipeline is triggered cdk diff and later cdk deploy pull the building image since the image not found locally

```
+ cdk diff
Bundling asset Root-dev/...[CUT].../Code/Stage...
Unable to find image '[123456789012.dkr.ecr.eu-central-1.amazonaws.com/com-python-build-image:latest](http://1234566789012.dkr.ecr.eu-central-1.amazonaws.com/com-python-build-image:latest)' locally
latest: Pulling from com-python-build-image
8740c948ffd4: Pulling fs layer
69038a8b17e6: Pulling fs layer
``` 

I'm just wondering, whether it is possible to put the bundling image to the custom image we use for the pipeline. :thinking: 
So, basically, it would be docker in docker.
